From c493301c7cf2ac0380e5f64e3c821989790071a9 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:32:10 -0400
Subject: [scsi] hpsa: fix non-x86 builds

Message-id: <1410186734-19986-80-git-send-email-thenzl@redhat.com>
Patchwork-id: 92075
O-Subject: [RHEL6.6 e-stor PATCH 79/83] hpsa: fix non-x86 builds
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

commit 28e134464734 " hpsa: enable unit attention reporting"
turns on unit attention notifications, but got the change wrong for
all architectures other than x86, which now store an uninitialized
value into the device register.

Gcc helpfully warns about this:

../drivers/scsi/hpsa.c: In function 'hpsa_set_driver_support_bits':
../drivers/scsi/hpsa.c:6373:17: warning: 'driver_support' is used uninitialized in this function [-Wuninitialized]
  driver_support |= ENABLE_UNIT_ATTN;
                 ^

This moves the #ifdef so only the prefetch-enable is conditional
on x86, not also reading the initial register contents.

Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Fixes: 28e134464734 " hpsa: enable unit attention reporting"
Cc: stable@vger.kernel.org # v3.14+
Acked-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 0b9e7b741f2bf8103b15bb14d5b4a6f5ee91c59a)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 9d23961..31d2614 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -6392,9 +6392,9 @@ static inline void hpsa_set_driver_support_bits(struct ctlr_info *h)
 {
  u32 driver_support;
 
-#ifdef CONFIG_X86
- /* Need to enable prefetch in the SCSI core for 6400 in x86 */
  driver_support = readl(&(h->cfgtable->driver_support));
+ /* Need to enable prefetch in the SCSI core for 6400 in x86 */
+#ifdef CONFIG_X86
  driver_support |= ENABLE_SCSI_PREFETCH;
 #endif
  driver_support |= ENABLE_UNIT_ATTN;
-- 
1.7.1